### DISCUSSION TITLE
docs/obs/tracing

### DISCUSSION THREAD

**Discussion by giscus** [gogf/gf-site#126] [General]
# docs/obs/tracing

GoFrame框架支持服务链路跟踪，通过实现标准化的分布式链路跟踪，帮助开发者监控和分析服务之间的交互过程。利用此功能可以有效提高系统的性能和可靠性，帮助定位问题源头，优化服务架构，保证系统在复杂环境中的高效运行。

https://goframe.org/docs/obs/tracing

**Comment by autothing** [comment]
你好，我直接按照示例，来接入trace。但是上报的时候会报错：traces export: failed to send to http://localhost:14268/api/traces: 400 Bad Request 。排查了好久了，一直接入失败，可以抽空看下吗

**Reply by autothing** [reply]
```go
package main

import (
	"context"

	"github.com/gogf/gf/contrib/trace/otlphttp/v2"
	"github.com/gogf/gf/v2/database/gdb"
	"github.com/gogf/gf/v2/frame/g"
	"github.com/gogf/gf/v2/net/ghttp"
	"github.com/gogf/gf/v2/os/gctx"
)

type cTrace struct{}

const (
	serviceName = "xxxx"
	endpoint    = "localhost:14268"
	path        = "api/traces"
)

func main() {
	var ctx = gctx.New()
	shutdown, err := otlphttp.Init(serviceName, endpoint, path)
	if err != nil {
		g.Log().Fatal(ctx, err)
	}
	defer shutdown(ctx)

	// Start HTTP server.
	s := g.Server()
	s.Use(ghttp.MiddlewareHandlerResponse)
	s.Group("/", func(group *ghttp.RouterGroup) {
		group.ALL("/user", new(cTrace))
	})
	s.SetPort(8199)
	s.Run()
}
```

**Reply by houseme** [reply]
根据安装的 `jaegertracing/all-in-one:1.55`的版本选择查看文档

建议看下如下文档：
v2: https://www.jaegertracing.io/docs/2.1/apis/
v1: https://www.jaegertracing.io/docs/1.64/apis/

V1版本HTTP使用的端口是：`4318`

V2 版本读写分离：
HTTP 写入使用的端口是：`4318`
HTTP 读取使用的端口是：`16686`

**Reply by autothing** [reply]
太感谢了🙏， 原来是端口和path问题。修改后可以显示了
